src/stratigraphy/benchmark/score.py

```diff
@@ -80,95 +55,38 @@ def get_scores(
-        document_level_metrics["recall"].append(recall)
-        document_level_metrics["F1"].append(f1(precision, recall))
-        document_level_metrics["Number Elements"].append(number_of_truth_values[filename])
-        document_level_metrics["Number wrong elements"].append(fn + fp)
-        try:
-            document_level_metrics["Depth_interval_accuracy"].append(depth_interval_hits / depth_interval_occurences)
-            depth_interval_accuracies.append(depth_interval_hits / depth_interval_occurences)
-        except ZeroDivisionError:
-            document_level_metrics["Depth_interval_accuracy"].append(None)
-
-    if len(document_level_metrics["precision"]):
-        overall_precision = sum(document_level_metrics["precision"]) / len(document_level_metrics["precision"])
-        overall_recall = sum(document_level_metrics["recall"]) / len(document_level_metrics["recall"])
-        try:
-            overall_depth_interval_accuracy = sum(depth_interval_accuracies) / len(depth_interval_accuracies)
-        except ZeroDivisionError:
-            overall_depth_interval_accuracy = None
-    else:
-        overall_precision = 0
-        overall_recall = 0
-
-    if overall_depth_interval_accuracy is None:
-        overall_depth_interval_accuracy = 0
-
-    if return_document_level_metrics:
-        return {
-            "F1": f1(overall_precision, overall_recall),
-            "precision": overall_precision,
-            "recall": overall_recall,
-            "depth_interval_accuracy": overall_depth_interval_accuracy,
-        }, pd.DataFrame(document_level_metrics)
-    else:
-        return {
-            "F1": f1(overall_precision, overall_recall),
-            "precision": overall_precision,
-            "recall": overall_recall,
-            "depth_interval_accuracy": overall_depth_interval_accuracy,
-        }
+        layer_metrics.metrics[filename] = Metrics(
+            tp=hits, fp=len(file_prediction.layers) - hits, fn=number_of_truth_values[filename] - hits
+        )
+        if depth_interval_occurences > 0:
+            depth_interval_metrics.metrics[filename] = Metrics(
+                tp=depth_interval_hits, fp=depth_interval_occurences - depth_interval_hits, fn=0
+            )
+
+    return layer_metrics, depth_interval_metrics
 
 
 def evaluate_borehole_extraction(
     predictions: dict[str, FilePredictions], number_of_truth_values: dict
-) -> tuple[dict, pd.DataFrame]:
+) -> dict[str, DatasetMetrics]:
     """Evaluate the borehole extraction predictions.
 
     Args:
        predictions (dict): The FilePredictions objects.
        number_of_truth_values (dict): The number of layer ground truth values per file.
 
     Returns:
-        tuple[dict, pd.DataFrame]: A tuple containing the overall metrics as a dictionary and the
-        individual document metrics as a DataFrame.
+        dict[str, DatasetMetrics]: A dictionary that maps a metrics name to the corresponding DatasetMetrics object
```

Comment:
I would tend to suggest creating a data object for this dict. This makes it clearer in my opinion. In the end, the dict will only contain filenames and their metrics, but by creating an object, we could make some methods to make the iterations through the object easier, for instance.

Reply:
Yes, good idea, I already thought about this as well. I don't see much potential yet to simplify the code much by doing this, but at least the evaluate() method in score.py can become a lot shorter, by moving the computation of the document_level_metrics dataframe and the metrics dictionary into the new class.